.evergreen/install-dependencies.sh

```diff
@@ -9,6 +9,10 @@ INSTALL_DIR="${PROJECT_DIRECTORY}/opt"
 BUILD_DIR="${PROJECT_DIRECTORY}/libmongoc-build"
 EVG_DIR=$(dirname $0)
 
+# this is set by drivers-matrix-testing, and it's a special variable used in swiftenv
+# leaving it set messes with the installation
+unset PLATFORM
```

Comment:
this shouldn't affect the test suite if it needs access to `$PLATFORM`, since we run `install-dependencies.sh` in its own shell